sysdig-operator/internal/controller/sysdigteamgo_controller.go

```diff
@@ -243,9 +246,29 @@ func (r *SysdigTeamGoReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	// Handle deletion: Check if the DeletionTimestamp is set
 	if !sysdigTeam.ObjectMeta.DeletionTimestamp.IsZero() {
+
+		// If the old finalizer exists in a SysdigTeam, remove it.
+		if containsString(sysdigTeam.ObjectMeta.Finalizers, sysdigTeamFinalizerOld) {
+			logger.Info("Removing old finalizer...")
+			sysdigTeam.ObjectMeta.Finalizers = removeString(sysdigTeam.ObjectMeta.Finalizers, sysdigTeamFinalizerOld)
+			if err := r.Update(ctx, &sysdigTeam); err != nil {
```

Comment:
General looks good to me, but I was planning to use this pageage's function instead of myown created function removeString to do it:
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil has a function called controllerutil.RemoveFinalizer


So instead of update twice, maybe we can run 
controllerutil.RemoveFinalizer(sysdigTeam.ObjectMeta.Finalizers, sysdigTeamFinalizerOld) 
 controllerutil.RemoveFinalizer(sysdigTeam.ObjectMeta.Finalizers, sysdigTeamFinalizer)
 
 And just signgle update:
 r.Update(ctx, &sysdigTeam)